xc/xenops_server_xen.ml

```diff
@@ -873,6 +873,49 @@ let upgrade_for_migration ~xc features =
   with Xenctrlext.Unix_error (Unix.ENOSYS, _) ->
     debug "xc_get_msr_arch_caps: ENOSYS"
 
+module BitSet = struct
+  type t = int64
+
+  (** [diff lhs rhs] computes all the features that are present in [lhs] but absent in [rhs] *)
+  let diff lhs rhs =
+    (* all bits that are present on lhs and missing on rhs:
+       = lhs & ~rhs *)
+    Int64.(logand lhs (lognot rhs))
+
+  let logor = Int64.logor
+end
+
+module Featureset = struct
+  (* an array of unsigned 32-bit numbers represented using int64 *)
+  type t = int64 array
```

Comment:
You could use `int32 array`

Reply:
Yes, we should probably change that in xenctrl too (although it wouldn't actually save memory on 64-bit architectures because allocations are aligned to word-size)